integration-tests/contracts/ValueCalls.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
+// @unsupported: evm
+pragma solidity >=0.7.0;
+
+contract ValueContext {
+    function getSelfBalance() external view returns(uint256) {
+        uint selfBalance;
+        assembly {
+            selfBalance := selfbalance()
+        }
+        return selfBalance;
+    }
+
+    function getBalance(
+        address _address
+    ) external payable returns(uint256) {
+        return _address.balance;
+    }
+
+    function getCallValue() public payable returns(uint256) {
+        return msg.value;
+    }
+}
+
+contract ValueCalls is ValueContext {
+
+    receive() external payable { }
+
```

Comment:
Do you have a test for a /nonpayable/ function reverting when value > 0?

Reply:
Added.